src/libraries/System.Collections.Immutable/tests/Frozen/FrozenDictionaryTests.cs

```diff
@@ -392,19 +406,68 @@ public void CreateHugeDictionary_Success(int largeCount)
         }
     }
 
-    public class FrozenDictionary_Generic_Tests_int_int : FrozenDictionary_Generic_Tests<int, int>
+    public abstract class FrozenDictionary_Generic_Tests_base_for_numbers<T> : FrozenDictionary_Generic_Tests<T, T>
     {
         protected override bool DefaultValueAllowed => true;
 
-        protected override KeyValuePair<int, int> CreateT(int seed)
+        protected override KeyValuePair<T, T> CreateT(int seed)
         {
             Random rand = new Random(seed);
-            return new KeyValuePair<int, int>(rand.Next(), rand.Next());
+            return new KeyValuePair<T, T>(Next(rand), Next(rand));
         }
 
-        protected override int CreateTKey(int seed) => new Random(seed).Next();
+        protected override T CreateTKey(int seed) => Next(new Random(seed));
+
+        protected override T CreateTValue(int seed) => CreateTKey(seed);
+
+        protected abstract T Next(Random random);
+    }
+
+
+    public class FrozenDictionary_Generic_Tests_int_int : FrozenDictionary_Generic_Tests_base_for_numbers<int>
+    {
+        protected override int Next(Random random) => random.Next(); 
+    }
+
+    public class FrozenDictionary_Generic_Tests_uint_uint : FrozenDictionary_Generic_Tests_base_for_numbers<uint>
+    {
+        protected override uint Next(Random random) => (uint)random.Next(int.MinValue, int.MaxValue);
+    }
+
+    public class FrozenDictionary_Generic_Tests_short_short : FrozenDictionary_Generic_Tests_base_for_numbers<short>
+    {
+        protected override bool AllowVeryLargeSizes => false;
+
+        protected override int MaxUniqueValueCount => short.MaxValue - short.MinValue;
+
+        protected override short Next(Random random) => (short)random.Next(short.MinValue, short.MaxValue);
+    }
+
+    public class FrozenDictionary_Generic_Tests_ushort_ushort : FrozenDictionary_Generic_Tests_base_for_numbers<ushort>
+    {
+        protected override bool AllowVeryLargeSizes => false;
+
+        protected override int MaxUniqueValueCount => ushort.MaxValue;
+
+        protected override ushort Next(Random random) => (ushort)random.Next(ushort.MinValue, ushort.MaxValue);
+    }
+
+    public class FrozenDictionary_Generic_Tests_byte_byte : FrozenDictionary_Generic_Tests_base_for_numbers<byte>
+    {
+        protected override bool AllowVeryLargeSizes => false;
+
+        protected override int MaxUniqueValueCount => byte.MaxValue;
+
+        protected override byte Next(Random random) => (byte)random.Next(byte.MinValue, byte.MaxValue);
+    }
+
+    public class FrozenDictionary_Generic_Tests_sbyte_sbyte : FrozenDictionary_Generic_Tests_base_for_numbers<sbyte>
```

Comment:
You stopped short of creating tests for nint and nuint?

Reply:
> You stopped short of creating tests for nint and nuint? Seems like those could be some of the more valuable ones, since the optimization is only relevant when they're 4 bytes.

It was an oversight. I added the tests and fixed the bug it had discovered (`IntPtr` started implementing `IComparable<IntPtr>` in .NET 5)